pinecone_text/dense/openai_encoder.py

```diff
@@ -76,9 +80,14 @@ def _encode(
             )
 
         try:
-            response = self._client.embeddings.create(
-                input=texts_input, model=self._model_name
+            params: Dict[str, Any] = dict(
+                input=texts_input,
+                model=self._model_name,
             )
+            if self._dimension is not None:
+                assert self._dimension > 0, "dimension must be a positive integer"
```

Comment:
1) I think it's better to check that on init and not here
2) Maybe it's better to use here `self.dimension` and not the private property